pkg/virt-controller/watch/migration.go

```diff
@@ -516,6 +528,8 @@ func (c *MigrationController) updateStatus(migration *virtv1.VirtualMachineInsta
 		}
 	}
 
+	c.flushPendingQueue(migration, migrationCopy)
```

Comment:
Let's say that there are X running migrations (on the main queue) and Y pending migrations, and that Y is much bigger than X.

In this case IIUC we'd return to the state where we have a queue that's blown out with pending migrations. They will, in turn, be enqueued again to the pending queue, but perhaps we can limit the amount of migrations we flush into the main queue in a better way.

For example, how about we enqueue `ParallelOutboundMigrationsPerNode`, or even double than that, just to ensure we don't blow up the main queue again?

Reply:
Yes, that was a compromise we made to avoid forgotten migrations... And it was still not enough, as you found!
Note that today we just keep re-enqueuing everything always, so this is still an improvement!
Maybe the added flush-pending-on-empty-queue will allow for partial flushes, I need to think about it!